server/server.go

```diff
@@ -66,24 +67,48 @@ type config struct {
 	Port             string `envconfig:"PORT" default:"6789"`
 	ReportPath       string `envconfig:"REPORT_PATH" default:""`
 	PathPrefix       string `envconfig:"KEPLOY_PATH_PREFIX" default:"/"`
+
+	// Use this option to tune the behaviour of the path where the logs would be written.
+	LogLocation models.LogLocation `envconfig:"LOG_LOCATION" default:"console"`
+
+	// The absolute or relative filepath of the files to store the logs in.
+	// This is platform agnostic, so forward and backward slashes both are fine.
+	// If the file doesn't exists, one will be created for you.
+	LogFileLocation string `envconfig:"LOG_FILE_LOCATION" default:"/tmp/keploy.log"`
 }
 
 func Server(ver string) *chi.Mux {
 	rand.Seed(time.Now().UTC().UnixNano())
 
+	// Parse the config before creating the logger to figure out whether we need
+	// to pipe the logs or not.
+	var conf config
+	err := envconfig.Process("keploy", &conf)
+	if err != nil {
+		fmt.Printf("failed to read/process configuration with error %v", err)
+		return nil
+	}
+
 	logConf := zap.NewDevelopmentConfig()
 	logConf.Level = zap.NewAtomicLevelAt(zap.InfoLevel)
+
+	// If the user has requested to the pipe the logs in a file,
+	// modify the internal config for zap which supports multiple syncers.
+	switch conf.LogLocation {
+	case models.Console:
+		logConf.OutputPaths = []string{"stderr"}
+	case models.File:
+		logConf.OutputPaths = []string{conf.LogFileLocation}
```

Comment:
Please create a file if already not exist for pipe and redirect the logs.

Reply:
Hi, @re-Tick  can you please elaborate? My code already handles this scenario, i.e, it creates the file if that file does not exist. (You just need to pass in the file names to conf, and the internals of `zap` would ensure file creation.

Reply:
It throws following error when there is no tmp directory in keploy repo.
```
panic: couldn't open sink "tmp/keploy.log": open tmp/keploy.log: no such file or directory

goroutine 1 [running]:
go.keploy.io/server/server.Server({0x1049c2144, 0x9})
        /Users/xxx/Desktop/go-practice/tmp/aerowisca/keploy/server/server.go:108 +0x1c98
main.main()
        /Users/xxx/Desktop/go-practice/tmp/aerowisca/keploy/cmd/server/main.go:14 +0x30
```

 Steps to reproduce:
1. Use this env variables and make sure that there is no tmp folder in the keploy repo
```bash
export KEPLOY_LOG_LOCATION=file
export LOG_FILE_LOCATION="tmp/keploy.log"
```
2. source the env variables and run the keploy server
```bash
source .env && go run cmd/server/main.go
```

Reply:
Yes, that understanding is correct, and is in line with what I've seen in other projects (i.e, external packages should only create missing files, instead of creating entire directories, since we do not know about the specific permissions of the nested directories that we are going to create).

It's the user's responsibility to specify a correct directory (regardless of file being present there). (In documentation, that's a typo, On Unix systems, I was planning to default it to `/tmp` directory).

I can of course add the feature for creating the nested directories as well, but just curious on what other's thoughts are. Do you think I should add it?

Reply:
@re-Tick  What about the other features (assuming the directory exists)? Did you get a chance to experiment with those by creating `tmp` folder? Let me know if any changes/modifications are needed there.

Reply:
Yes, I have tried after creating the tmp folder, and it is working.

Reply:
I think we should create the missing directory or nested directory because this will decrease a step of creating the nested directory and be more quick.

Reply:
Makes sense. I've added the feature to create nested directories in my new commit. Can you please do one more round of experiments at your end?

Yes, the current implementation should handle absolute as well as relative paths.